nodes/modifier_make/isoLine2D.py

```diff
@@ -0,0 +1,694 @@
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+import itertools
+from collections import defaultdict
+from math import sin, cos, pi, degrees, radians ,atan2, asin
+from mathutils.geometry import intersect_line_line as LineIntersect
+from mathutils import Vector
+
+import bpy
+import bmesh
+from bpy.props import BoolProperty, IntProperty, FloatProperty, EnumProperty
+
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import (fullList, match_long_repeat,dataCorrect, repeat_last, updateNode, match_long_cycle)
+from sverchok.utils.sv_mesh_utils import mesh_join
+from sverchok.utils.cad_module_class import CAD_ops
+from sverchok.utils.sv_bmesh_utils import bmesh_from_pydata
+
+modeItems = [("Constant", "Constant", ""), ("Weighted", "Weighted", "")]
+listMatchItems = [("Long Repeat", "Long Repeat", ""), ("Long Cycle", "Long Cycle", "")]
+def order_points(edge, point_list):
+    ''' order these edges from distance to v1, then
+    sandwich the sorted list with v1, v2 '''
+    v1, v2 = edge
+    dist = lambda co: (v1-co).length
+    point_list = sorted(point_list, key=dist)
+    return [v1] + point_list + [v2]
+
+
+def remove_permutations_that_share_a_vertex(cm, bm, permutations):
+    ''' Get useful Permutations '''
+
+    final_permutations = []
+    for edges in permutations:
+        raw_vert_indices = cm.vertex_indices_from_edges_tuple(bm, edges)
+        if cm.duplicates(raw_vert_indices):
+            continue
+
+        # reaches this point if they do not share.
+        final_permutations.append(edges)
+
+    return final_permutations
+
+
+def get_valid_permutations(cm, bm, edge_indices):
+    raw_permutations = itertools.permutations(edge_indices, 2)
+    permutations = [r for r in raw_permutations if r[0] < r[1]]
+    return remove_permutations_that_share_a_vertex(cm, bm, permutations)
+
+
+def can_skip(cm, closest_points, vert_vectors):
+    '''this checks if the intersection lies on both edges, returns True
+    when criteria are not met, and thus this point can be skipped'''
+    if not closest_points:
+        return True
+    if not isinstance(closest_points[0].x, float):
+        return True
+    if cm.num_edges_point_lies_on(closest_points[0], vert_vectors) < 2:
+        return True
+
+    # if this distance is larger than than VTX_PRECISION, we can skip it.
+    cpa, cpb = closest_points
+    return (cpa-cpb).length > cm.VTX_PRECISION
```

Comment:
the downside to using the static `cm.VTX_PRECISION` is that it will behave differently if the input geometry is thousands of units in scale.. ( range 1000, 10 000 ).. isntead of the usual scene scale (0.0... 10.0 ) -- in EdgesIntersection node I think i made a manual 'scale' parameter accessible from the `draw_buttons_ext` area.